browser/src/Services/QuickOpen/VSCodeFilter.ts

```diff
@@ -49,7 +50,16 @@ export const vsCodeFilter = (
             ? listOfSearchTerms.slice(1).join("") + listOfSearchTerms[0]
             : listOfSearchTerms[0]
 
-    const filteredOptions = processSearchTerm(vsCodeSearchString, options)
+    // Adds a cache for the scores. This is needed to stop the final score
+    // compare from repeating all the scoring logic again.
+    // Currently, this only persists for the current search, which will speed
+    // up that search only.
+    // TODO: Is it worth instead persisting this cache?
```

Comment:
Good question - do you mean persisting this cache on the filesystem, so that it's available on startup?  Or persisting in memory for the duration of the session? I think the latter may make sense... the former might be overkill for now (but maybe something we could do later)

Reply:
Or we could stick with the current strategy for now - it's at least much faster since we don't repeat the scoring logic for final scoring, and hook up the persistence later (if needed)

Reply:
I was meaning the later, persisting for the session.
It would make subsequent searches pretty much instant, The only concern I had is that every single search is persisted so it could end up getting quite big.

That is for `QuickOpen` the results are stored for `Q`, `Qu`, `Qui`...and so on. I think VSCode does persist it for the whole session though.